test/models/test_models.py

```diff
@@ -126,3 +128,49 @@ def test_roberta_bundler_from_config(self):
             encoder_state_dict['encoder.' + k] = v
         model = torchtext.models.RobertaModelBundle.from_config(encoder_conf=dummy_encoder_conf, head=dummy_classifier_head, checkpoint=encoder_state_dict)
         self.assertEqual(model.state_dict(), dummy_classifier.state_dict())
+
+    def test_roberta_bundler_train(self):
```

Comment:
Since this is a rather complicated for unit test, can you add doc string?
There are many style for writing docstring for tests, but one should be able to describe a test like "Given A, B should be met."

If the purpose of the test is just ensuring no error happens, then I call that kind of tests as "smoke test". (Could be different from what the "smoke test" means in industry though)

Reply:
Thanks @mthrok, I agree on all the points mentioned. This would go in integration/smoke tests. Let me organize all the model tests in follow-up PR. There are other model test (not introduced in this PR) too that accordingly need to be separated.